In [40]:
import pandas as pd
import os
import kagglehub

path = kagglehub.dataset_download("patrickb1912/ipl-complete-dataset-20082020")
deliveries = pd.read_csv(os.path.join(path, "deliveries.csv"), nrows=50000)
df= deliveries


In [41]:
df.columns

Index(['match_id', 'inning', 'batting_team', 'bowling_team', 'over', 'ball',
       'batter', 'bowler', 'non_striker', 'batsman_runs', 'extra_runs',
       'total_runs', 'extras_type', 'is_wicket', 'player_dismissed',
       'dismissal_kind', 'fielder'],
      dtype='str')

Q1 — Most Runs

In [42]:
most_runs= df.groupby('batter')['batsman_runs'].sum().sort_values(ascending=False).reset_index()
most_runs.columns = ['Batsman_name','total_runs']
print(most_runs.head(5))


   Batsman_name  total_runs
0      SK Raina        1536
1  SR Tendulkar        1446
2  AC Gilchrist        1397
3     JH Kallis        1376
4     RG Sharma        1373


Insight: Sk Raina(Suresh Raina) is the most consistent run scorer in the sampled IPL data. His volume suggests he batted in high pressure middle order positions across multiple seasons rather than one standout year.

Q2 — Most Wickets

In [43]:
most_wickets =df.groupby('bowler')['is_wicket'].sum().sort_values(ascending=False).reset_index()
most_wickets.columns = ['Bowler', 'Total_wickets']
print(most_wickets.head(5))

      Bowler  Total_wickets
0  IK Pathan             64
1   RP Singh             63
2  JA Morkel             57
3    PP Ojha             56
4   SK Warne             56


Insight: K Pathan leads with 64 wickets in the sampled data, closely followed by RP Singh with 63 with both consistently taking wickets across multiple seasons rather than peaking in one.

Q3 — Most Dot Balls

In [44]:
most_dots = df[df['total_runs'] == 0].groupby('bowler').size().sort_values(ascending=False).reset_index()
most_dots.columns = ['Bowler', 'Dot_balls']
most_dots.head(5)


,Bowler,Dot_balls
0,RP Singh,462
1,IK Pathan,444
2,M Muralitharan,431
3,DW Steyn,390
4,SK Warne,387


Insight: RP Singh bowled the most dot balls (462) in the sample. Combined with his second place wicket tally, this confirms he was IPL's most economical as well as a threatening bowler and suffocating batters while consistently taking wickets.